### PR TITLE
Now every user manages its own dbdirect IPs

### PR DESCRIPTION
**Pull request by thedae**
Sync script to be executed in bch, this will clear redis and dump the dbdirect_ips from org owner to all users that belong to that org:

```ruby
Carto::User.find_each do |user|
  $users_metadata.DEL('dbdirect:' + user.username)
  if user.organization.present?
    if user.organization.owner != user
      user.dbdirect_effective_ips = user.organization.owner.dbdirect_effective_ips
      user.save!
    end
  end
  if user.dbdirect_effective_ips.any?
    $users_metadata.HSETNX('dbdirect:' + user.username, 'ips', user.dbdirect_effective_ips.join(','))
  end
end
```

### CODE REVIEW

**Review by alasarr — APPROVED** [review]

File: app/models/carto/user.rb [hunk 1]
```diff
@@ -320,11 +320,7 @@ def dbdirect_effective_ip
   end
 
   def dbdirect_bearer
-    if organization.present? && organization.owner != self
-      organization.owner.reload
-    else
-      reload
-    end
+    reload
   end
```
Comment (amiedes) [inline app/models/carto/user.rb:324]: Given that `dbdirect_bearer` is now the same as `self.reload`, I think that:

```ruby
def dbdirect_effective_ips=(ips)
  ips ||= []
  bearer = dbdirect_bearer
  if bearer.dbdirect_ip
    bearer.dbdirect_ip.update!(ips: ips)
  else
    bearer.create_dbdirect_ip!(ips: ips)
  end
end

def dbdirect_effective_ip
  dbdirect_bearer.dbdirect_ip
end

def dbdirect_bearer
  reload
end
```

Could be rewritten as:

```ruby
def dbdirect_effective_ips=(ips = [])
  reload
  dbdirect_ip ? dbdirect_ip.update!(ips: ips) : create_dbdirect_ip!(ips: ips)
end

def dbdirect_effective_ip
  reload.dbdirect_ip
end
```

Could you check in local if it works as expected and apply it if relevant?

**Review by amiedes — CHANGES_REQUESTED** [review]
Nice work 🙂

I've just left a suggestion about some code that I think can be refactored a bit.

File: app/models/carto/user.rb (hunk 1 above)
Comment (thedae) [reply app/models/carto/user.rb:324]: Sure, I didn't want to remove the bearer logic just in case but I guess we can revert whenever we want :)

Comment (thedae) [reply app/models/carto/user.rb:316]: Fixed, and seems to work. Thanks!

**Review by amiedes — APPROVED** [review]
🚀

**Pull request merged**